src/lib/geojson.js

```diff
@@ -35,3 +35,41 @@ export function expandMultiGeometry(
 
   return { result, rejected };
 }
+
+export function immutablyReplaceCoordinate(
```

Comment:
Nit: Add comment header describing the intended functionality. 

Nit: Naming, this seems to focus on closing polygons. Maybe `closePolygons(geojson)`, or `normalizeGeojson(geojson)`.

You could also offer destructive, faster versions (`closePolygonsInPlace` or `normalizeGeojsonInPlace()`

Reply:
Added a comment. It is for multiple types of geometries. This is basically the equivalent of immutable.js's `setIn` function.

It is the immutable version of `coordinates[a][b][c] = updatedCoordinate`. You can do `const updated = immutablyReplaceCoordinate(coordinates, [a, b, c], updatedCoordinate)`